src/intervals/interval_operations/set_operations.jl

```diff
@@ -0,0 +1,99 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+#=  This file contains the functions described in sections 9.3 of the
+    IEEE Std 1788-2015 (Set operations) and required for set-based flavor
+    in section 10.5.7. Some other related functions are also present.
+=#
+
+"""
+    intersect(a, b)
+    ∩(a,b)
+
+Returns the intersection of the intervals `a` and `b`, considered as
+(extended) sets of real numbers. That is, the set that contains
+the points common in `a` and `b`.
+
+Implement the `intersection` function of the IEEE Std 1788-2015 (section 9.3).
+"""
+function intersect(a::Interval{T}, b::Interval{S}) where {T, S}
+    isdisjoint(a, b) && return emptyinterval(promote_type(T, S))
+    return Interval{promote_type(T, S)}(max(a.lo, b.lo), min(a.hi, b.hi))
+end
+
+function intersect(a::Complex{F}, b::Complex{F}) where {F<:Interval}
+    isdisjoint(a, b) && return emptyinterval(Complex{F})
+    return complex(intersect(real(a), real(b)), intersect(imag(a), imag(b)))
+end
+
+"""
+    intersect(a::Interval{T}...) where T
+
+Return the n-ary intersection of its arguments.
+
+This function is applicable to any number of input intervals, as in
+`intersect(a1, a2, a3, a4)` where `ai` is an interval.
+If your use case needs to splat the input, as in `intersect(a...)`, consider
+`reduce(intersect, a)` instead, because you save the cost of splatting.
+"""
+function intersect(a::F...) where {F<:Interval}
+    low = maximum(broadcast(ai -> ai.lo, a))
+    high = minimum(broadcast(ai -> ai.hi, a))
```

Comment:
Maybe the following helps: We may avoid here the anonymous functions by using `inf` and `sup` and broadcasting directly:
```julia
low = maximum(inf.(a))
high = minimum(sup.(a))
```